lib/galaxy/datatypes/binary.py

```diff
@@ -1528,6 +1532,35 @@ def display_peek(self, dataset):
         except Exception:
             return "HDF5 Model (%s)" % (nice_size(dataset.get_size()))
 
+    def display_data(self, trans, dataset, preview=False, filename=None, to_ext=None, **kwd):
+        preview = util.string_as_bool(preview)
+
+        if to_ext or not preview:
+            to_ext = to_ext or dataset.extension
+            return self._serve_raw(trans, dataset, to_ext, **kwd)
+
+        rval = {}
+        try:
+            with h5py.File(dataset.file_name, "r") as handle:
+                rval['Attributes'] = {}
+                for k in handle.attrs.keys():
```

Comment:
I think you don't need `keys()` here: `h5py.AttributeManager` supports a dict interface, and by doing `for k in handle.attrs` you'll be iterating over the keys (see the `__iter__` method in the [docs](https://docs.h5py.org/en/latest/high/attr.html#reference))